src/hooks/useDisputableVotes.js

```diff
@@ -133,16 +136,40 @@ async function processVote(vote, apps) {
     naysPct: vote.naysPct,
     yeasPct: vote.yeasPct,
     status: vote.status,
+    target: getTargetData(apps, disputableVotingApp.address),
   }
 
   if (vote.script === EMPTY_SCRIPT) {
     return extendedVote
   }
 
   const description = await describeScript(vote.script, apps)
+  if (description[0] && description[0].to) {
+    extendedVote.target = {
+      address: description[0].to,
+      name: description[0].name || description[0].identifier || '',
+      icon: '',
+    }
+  }
+
+  if (
+    apps &&
+    apps.filter((app) => app.address === extendedVote.target.address).length > 0
+  ) {
+    extendedVote.target = getTargetData(apps, extendedVote.target.address)
+  }
```

Comment:
Find this logic a little bit confusing, perhaps we could clarify the intention by assigning these as descriptive variables / adding comments? Eventually we might want to break them up into composable functions but in the interest of time just a little bit more description here would be useful.

Reply:
i tried to explain this a little bit more. let me know if this is ok :)